compiler/fir/fir2ir/src/org/jetbrains/kotlin/fir/backend/Fir2IrLocalStorage.kt

```diff
@@ -49,6 +47,10 @@ class Fir2IrLocalStorage {
         return localClassCache[localClass]
     }
 
+    fun getLocalClass(classId: ClassId): IrClass? {
+        return localClassCache.keys.find { firClass -> firClass.classId == classId }?.let { localClassCache[it] }
```

Comment:
I think it's a bit faster to search for entry from the beginning:

```
        return localClassCache.entries.find { (firClass, _) -> firClass.classId == classId }?.value
```

Reply:
Of course! 🤦  Thanks!